test/src/userobjects/PeriodicNodeMapTester.C

```diff
@@ -0,0 +1,92 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PeriodicNodeMapTester.h"
+#include "NonlinearSystemBase.h"
+
+registerMooseObject("MooseTestApp", PeriodicNodeMapTester);
+
+template <>
+InputParameters
+validParams<PeriodicNodeMapTester>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  params.addClassDescription("Test MooseMesh::buildPeriodicNodeMap()");
+  params.addCoupledVar("v", "coupled variable (should be periodic)");
+  return params;
+}
+
+PeriodicNodeMapTester::PeriodicNodeMapTester(const InputParameters & parameters)
+  : ElementUserObject(parameters),
+    _v_var(coupled("v")),
+    _dim(_mesh.dimension()),
+    _perf_buildmap(registerTimedSection("buildPeriodicNodeMap", 1))
+{
+  // collect mesh periodicity data
+  for (unsigned int i = 0; i < _dim; ++i)
+  {
+    if (!_mesh.isRegularOrthogonal() || !_mesh.isTranslatedPeriodic(_v_var, i))
+      paramError("v", "Variable must be periodic in all directions");
+
+    _periodic_min[i] = _mesh.getMinInDimension(i);
+    _periodic_max[i] = _mesh.getMaxInDimension(i);
+  }
+}
+
+void
+PeriodicNodeMapTester::initialize()
+{
+  // get a fresh point locator
+  auto point_locator = _mesh.getPointLocator();
+
+  // Get a pointer to the PeriodicBoundaries buried in libMesh
+  auto pbs = _fe_problem.getNonlinearSystemBase().dofMap().get_periodic_boundaries();
+
+  // rebuild periodic node map (this is the heaviest part by far)
+  std::multimap<dof_id_type, dof_id_type> periodic_node_map;
+  {
+    TIME_SECTION(_perf_buildmap);
```

Comment:
Unecessary. There's already a timer in this method.

Reply:
But it has level set to 5. I added a custom timer at level 1 to get an uncluttered per graph output with the default level